crates/types/src/identifiers.rs

```diff
@@ -94,18 +94,23 @@ impl From<LeaderEpoch> for crate::protobuf::common::LeaderEpoch {
 )]
 #[repr(transparent)]
 #[serde(transparent)]
-pub struct PartitionId(u64);
+pub struct PartitionId(u16);
+
+impl From<PartitionId> for u32 {
+    fn from(value: PartitionId) -> Self {
+        value.0 as u32
```

Comment:
If we write `u32::from(value.0)`, the compiler will tell us if we change the underlying `PartitionId` type to something that is not safely transformable.

Reply:
Great call. I like that.